File: server/src/main/java/com/fluffy/exam/domain/ExamDescription.java
package com.fluffy.exam.domain;

import com.fluffy.global.exception.BadRequestException;
import jakarta.persistence.Column;
import jakarta.persistence.Embeddable;
import lombok.AccessLevel;
import lombok.Getter;
import lombok.NoArgsConstructor;

@Embeddable
@NoArgsConstructor(access = AccessLevel.PROTECTED)
@Getter
public class ExamDescription {

    private static final int MAX_DESCRIPTION_LENGTH = 2000;

    @Column(name = "description", nullable = false)
    private String value;

    public ExamDescription(String value) {
        validateDescriptionLength(value);

        this.value = value;
    }

    private void validateDescriptionLength(String value) {
        if (value == null) {
            throw new BadRequestException("시험 설명은 필수입니다.");
        }

        if (value.length() > MAX_DESCRIPTION_LENGTH) {
            throw new BadRequestException("시험 설명은 %d자 이하여야 합니다.".formatted(MAX_DESCRIPTION_LENGTH));
        }
    }
}

File: server/src/main/kotlin/com/fluffy/exam/domain/ExamDescription.kt
package com.fluffy.exam.domain

import com.fluffy.global.exception.BadRequestException
import jakarta.persistence.Column
import jakarta.persistence.Embeddable

@Embeddable
data class ExamDescription(

    @field:Column(name = "description", nullable = false)
    val value: String
) {

    companion object {

        private const val MAX_DESCRIPTION_LENGTH = 2000

        fun empty() = ExamDescription("")
    }

    init {
        validateDescriptionLength()
    }

    private fun validateDescriptionLength() {
        if (value.length > MAX_DESCRIPTION_LENGTH) {
            throw BadRequestException("시험 설명은 ${MAX_DESCRIPTION_LENGTH}자 이하여야 합니다.")
        }
    }
}